src/toolchain/rust.rs

```diff
@@ -42,6 +49,30 @@ pub struct XtensaRust {
 }
 
 impl XtensaRust {
+    /// Get the latest version of Xtensa Rust toolchain.
+    pub fn get_latest_version() -> Result<String> {
+        let mut headers = header::HeaderMap::new();
+        headers.insert("Accept", "application/vnd.github.v3+json".parse().unwrap());
+
+        let client = reqwest::blocking::Client::builder()
+            .redirect(reqwest::redirect::Policy::none())
+            .user_agent("foo")
```

Comment:
It's probably fine for now (and really just not that big of a deal), but it'd be a good idea to set an actual user agent here 😁